components/entities/api/src/main/java/org/phenotips/entities/PrimaryEntity.java

```diff
@@ -53,7 +55,14 @@
      *
      * @return a valid document reference
      */
-    DocumentReference getDocument();
+    DocumentReference getDocumentReference();
+
+    /**
+     * Returns the document where the entity is stored.
+     *
+     * @return a valid document reference
+     */
+    XWikiDocument getDocument();
```

Comment:
I agree that downstream code will be affected, but in a simple search-and-replace manner, as ALL existing `getDocument()`-s return a `documentReference`. PT itself is small, and for now all the derivative code bases are small, and all can just user search-and-replace.

...and it is one of the most confusing things in our APIs, so IMHO we should just bite the bullet and rename everything. As @sergiu explained, we avoided returning XWikiDocument because XWiki was supposed to get rid of this class/refactor it, but it never happened in the last ~5 years, so why not take a chance and make it less confusing now, that we have to refactor a lot of related code anyway. It was a bad decision to start with, and IMHO better to fix it than to stick with it for compatibility reasons, when compatibility can easily be fixed (and it takes about the same amount of work both ways, as @itaiGershtansky already fixed things one way on the PT side)